### PR TITLE
OCM-9974 | fix: Change options validations to allow 0 min replicas as well

### PR DESCRIPTION
**Pull request by hunterkepley**
In the fix for [OCM-9974](https://issues.redhat.com//browse/OCM-9974), I had forgotten to also fix the options validations. This change is included here + updated tests

### CODE REVIEW

File: cmd/edit/machinepool/options.go [hunk 1]
```diff
@@ -64,11 +64,11 @@ func (m *EditMachinepoolOptions) Bind(args *EditMachinepoolUserOptions, argv []s
 	}
 
 	if m.args.autoscalingEnabled {
-		if m.args.minReplicas <= 0 {
-			return fmt.Errorf("Min replicas must be greater than zero when autoscaling is enabled")
+		if m.args.minReplicas < 0 {
+			return fmt.Errorf("Min replicas must be a positive number when autoscaling is enabled")
```
Comment (davidleerh) [inline cmd/edit/machinepool/options.go:68]: should it be worded, "must be a number that is 0 or greater?"

**Comment by lucasponce** [comment]
Is this fix for both ROSA Classic and HCP ? or only for Classic?

**Comment by den-rgb** [comment]
> Is this fix for both ROSA Classic and HCP ? or only for Classic?

It's for both

**Review by robpblake — APPROVED** [review]
/approve

**Comment by lucasponce** [comment]
But ROSA HCP doesn't support min-replicas to 0, it will fail from backend, which it could be ok

**Comment by hunterkepley** [comment]
@lucasponce Is there a list of differences for HCP and classic machinepool replica count somewhere? It seems QE and engineering have a disconnect on this, which has caused back and forth changes which could have been done in one change

**Comment by lucasponce** [comment]
@hunterkepley the available documentation is typically high-level, and some of the details are part of the API itself.

On this case, this is a limitation/feature from the Hypershift API one, in the case of Cluster Service, this is available in the API tests.

https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/cmd/clusters-service/servecmd/apiserver/node_pool_handlers_test.go?ref_type=heads#L485

**Pull request merged**